File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/usecase/GetCommentsUseCase.java
package com.example.postscommentsapp.domain.usecase;

import com.example.postscommentsapp.domain.model.Comment;
import com.example.postscommentsapp.domain.repository.CommentsRepository;

import java.util.List;

import javax.inject.Inject;

public class GetCommentsUseCase {

    private final CommentsRepository repository;

    @Inject
    public GetCommentsUseCase(CommentsRepository repository) {
        this.repository = repository;
    }

    public List<Comment> execute(int postId) {
        return repository.getCommentsByPost(postId);
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/usecase/GetCommentsUseCase.kt
package com.example.postscommentsapp.domain.usecase

import com.example.postscommentsapp.domain.model.Comment
import com.example.postscommentsapp.domain.repository.CommentsRepository
import javax.inject.Inject

class GetCommentsUseCase @Inject constructor(
    private val repository: CommentsRepository
) {
    suspend fun execute(postId: Int): List<Comment> = repository.getCommentsByPost(postId)
}
